docs/adr/0241-opinionated-api-changes.md

```diff
@@ -0,0 +1,76 @@
+
+## Context
+
+Flintlock requires users to specify all aspects of the MicroVM spec. If something (e.g. metadata network interface) is not included in the MicroVM spec, then it won't be created. This proposal details some possible opinionated defaults for MicroVM specifications in cases where an explicit configuration is not provided by the user.
+
+### Scope
+
+The scope of this proposal is not to decide on what aspects of MicroVM specifications are immutable / non-configurable. Rather it is to explore which potential aspects of the MicroVM specification can be provided by default, but also overridden/customised by the user if they so wish to.
+
+### Default Configurations
+
+Below are some proposed default configurations, along with default values where relevant.
+
+| Name | Description | Model | Value |
+|----------|---|---------|---------|
+| MicroVM Resources <img width="150px"/> | Provide default values for MicroVM VCPUs and RAM. This can be difficult to get right, but is even supported in Firecracker itself. However, Firecracker defaults to 1 VCPU and 128MB of RAM, which is probably a bit small for our particular use case. | `MicroVM.VCPU`<br />`MicroVM.MemoryInMb` <img width="400px"/> | `vcpu: 2`<br />`memory_in_mb: 1024` <img width="400px"/> |
+| Default Namespace | If a namespace is not provided than default to using the `default` namespace | `MicroVM.Namespace` | `default` |
+| Metadata network interface | Automatically generate metadata network interface with address of `169.254.0.1/16` and `allow_metadata_req: true` | `MicroVM.NetworkInterfaces` | `169.254.0.1/16` |
+| Remove allow metadata requests  | If the metadata network interface is created automatically then we should remove the `allow_metadata_req` option on the network interface model | `MicroVM.NetworkInterfaces` |  |
+| Network interface name | Automatically generate the network interface name when one is not provided. | `MicroVM.NetworkInterfaces[*].GuestDeviceName` | |
+| `RootVolume` and `AdditionalVolumes[]` | Separate the `Volumes` field in the model into two fields. `RootVolume` specifically for the root volume, allowing us to hide the `is_root` flag, and `AdditionalVolumes[]` for non-root volumes. | `MicroVM.Volumes` | `MicroVM.RootVolume` and `MicroVM.AdditionalVolumes[]` |
+
+### Unknowns/Discussion
+
+- ~~Can we generate volume names?~~
+- ~~If a volume is marked as `is_root: true`, can we automatically mount it at `/` if `mount_point` is not specified?~~
+
+### Request Example
+
+With the defaults as described in Default Configurations, a minimal MicroVM create request would look as follows:
+
+```json
+{
+  "microvm": {
+    "id": "mvm1",
+    "kernel": {
+      "image": "docker.io/richardcase/ubuntu-bionic-kernel:0.0.11",
+      "filename": "vmlinux"
+    },
+    "initrd": {
+      "image": "docker.io/richardcase/ubuntu-bionic-kernel:0.0.11",
+      "filename": "initrd-generic"
+    },
+    "interfaces": [
+      {
+        "type": 0 // macvtap
```

Comment:
A comment in the example is a bit misleading. At least until everyone start using [JSON5](https://json5.org/) (so like 20 years from now).